extensions/stackdriver/stackdriver.cc

```diff
@@ -151,12 +151,43 @@ std::string getMonitoringEndpoint() {
   return monitoring_endpoint;
 }
 
+// Get GCP project number.
+std::string getProjectNumber() {
+  std::string project_number;
+  if (!getValue({"node", "metadata", "PLATFORM_METADATA", kGCPProjectNumberKey},
+                &project_number)) {
+    return "";
+  }
+  return project_number;
+}
+
 void clearTcpMetrics(::Wasm::Common::RequestInfo& request_info) {
   request_info.tcp_connections_opened = 0;
   request_info.tcp_sent_bytes = 0;
   request_info.tcp_received_bytes = 0;
 }
 
+// Get local node metadata. If mesh id is not filled or does not exist,
+// fall back to default format `proj-<project-number>`.
+void getLocalNodeMetadata(google::protobuf::Struct* node_metadata) {
+  if (!getMessageValue({"node", "metadata"}, node_metadata)) {
+    return;
+  }
+  const auto mesh_id_it = node_metadata->fields().find("MESH_ID");
```

Comment:
Is there any default value for the MESH_ID here?

Reply:
It will just be empty string if not set.